### PR TITLE
Add Complex.real and Complex.imag

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
This commit adds the Complex.real and Complex.imag functions,
which return the real and imaginary part of a complex number
respectively.

See https://github.com/elixir-nx/nx/issues/693#issuecomment-1089203385.

### CODE REVIEW

**Comment by Benjamin-Philip** [comment]
Once we merge this, we'll also need to release a new complex version 0.3.1 to add Nx.real and Nx.imag.

**Comment by Benjamin-Philip** [comment]
@polvalente, and @josevalim could you review please? (I couldn't request you to review via the GitHub UI, hence the comment.)

File: lib/complex.ex [hunk 1]
```diff
@@ -456,6 +456,54 @@ defmodule Complex do
...
+
+      iex> Complex.imag(1)
+      0
+  """
+
+  @spec imag(t) :: number
+  @spec imag(number) :: number
+  def imag(z)
+
+  def imag(n) when is_number(n), do: 0
```
Comment (josevalim) [inline lib/complex.ex]:
```suggestion
  def imag(n) when is_number(n), do: n * 0
```

So floats still return floats.

**Comment by josevalim** [comment]
Thank you @Benjamin-Philip! I dropped a comment but i actually don't think this would be used by Nx because we can do much more efficient implementations, see my comments here: https://github.com/elixir-nx/nx/pull/711/files

So I would suggest to close this for now. :)

**Comment by Benjamin-Philip** [comment]
Do you think this would be useful in long pipes? If not, I'll go ahead and close it.

**Comment by josevalim** [comment]
Actually,even if we don't use it, it can be a good API.

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**